odoo/models.py

```diff
@@ -6205,7 +6205,7 @@ def diff(self, other, force=False):
         # call, 'names' only contains fields with a default. If 'self' is a new
         # line in a one2many field, 'names' also contains the one2many's inverse
         # field, and that field may not be in nametree.
-        todo = list(names) + list(nametree) if first_call else list(names)
+        todo = list(set(names) | set(nametree)) if first_call else list(names)
```

Comment:
The function `set` is good for deduplicating, but it makes the order non-deterministic.
I suggest to use `unique` instead, because it keeps items in first occurrence order.
```suggestion
        todo = list(unique(itertools.chain(names, nametree))) if first_call else list(names)
```